🐛 BUG: Could not resolve "_worker.js" when running `wrangler pages dev` with `--proxy` flag

### Which Cloudflare product(s) does this pertain to?

Wrangler core

### What version(s) of the tool(s) are you using?

3.22.1 [Wrangler]

### What version of Node are you using?

18.17.1

### What operating system are you using?

Mac

### Describe the Bug

I have a `_worker.js` in my project root:
```javascript=
export default {
  async fetch(request, env) {
    return new Response('Ok');
  },
}
```

I ran `pnpm exec wrangler pages dev --proxy <port>` and got:

```
✘ [ERROR] 1 error(s) and 0 warning(s) when compiling Worker.


▲ [WARNING] Failed to bundle _worker.js. Error: Build failed with 1 error:

  .wrangler/tmp/bundle-RCsC8p/middleware-insertion-facade.js:1:43: ERROR: Could not resolve
  "_worker.js"
      at failureErrorWithLog
  (/Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1636:15)
      at
  /Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1048:25
      at
  /Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1512:9
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    errors: [
      {
        detail: undefined,
        id: '',
        location: [Object],
        notes: [Array],
        pluginName: '',
        text: 'Could not resolve "_worker.js"'
      }
    ],
    warnings: []
  }
```

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

```
✘ [ERROR] 1 error(s) and 0 warning(s) when compiling Worker.


▲ [WARNING] Failed to bundle _worker.js. Error: Build failed with 1 error:

  .wrangler/tmp/bundle-RCsC8p/middleware-insertion-facade.js:1:43: ERROR: Could not resolve
  "_worker.js"
      at failureErrorWithLog
  (/Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1636:15)
      at
  /Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1048:25
      at
  /Users/matt/Library/pnpm/global/5/.pnpm/esbuild@0.17.19/node_modules/esbuild/lib/main.js:1512:9
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    errors: [
      {
        detail: undefined,
        id: '',
        location: [Object],
        notes: [Array],
        pluginName: '',
        text: 'Could not resolve "_worker.js"'
      }
    ],
    warnings: []
  }
```

--- a/.changeset/eighty-hotels-train.md
+++ b/.changeset/eighty-hotels-train.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+make sure the script path is correctly resolved in `pages dev` when no directory is specified

--- a/fixtures/pages-dev-proxy-with-script/_worker.js
+++ b/fixtures/pages-dev-proxy-with-script/_worker.js
@@ -1,0 +1,5 @@
+export default {
+	fetch() {
+		return new Response("hello from _worker.js");
+	},
+};

--- a/fixtures/pages-dev-proxy-with-script/custom/script/path/index.js
+++ b/fixtures/pages-dev-proxy-with-script/custom/script/path/index.js
@@ -1,0 +1,5 @@
+export default {
+	fetch() {
+		return new Response("hello from custom/script/path/index.js");
+	},
+};

--- a/fixtures/pages-dev-proxy-with-script/package.json
+++ b/fixtures/pages-dev-proxy-with-script/package.json
@@ -1,0 +1,21 @@
+{
+	"name": "pages-dev-proxy-with-script",
+	"version": "0.0.0",
+	"private": true,
+	"sideEffects": false,
+	"scripts": {
+		"dev": "npx wrangler pages dev public",
+		"test": "vitest run",
+		"test:watch": "vitest",
+		"type:tests": "tsc -p ./tests/tsconfig.json"
+	},
+	"devDependencies": {
+		"@cloudflare/workers-tsconfig": "workspace:*",
+		"@cloudflare/workers-types": "^4.20221111.1",
+		"undici": "^5.9.1",
+		"wrangler": "workspace:*"
+	},
+	"engines": {
+		"node": ">=16.13"
+	}
+}

--- a/fixtures/pages-dev-proxy-with-script/tests/index.test.ts
+++ b/fixtures/pages-dev-proxy-with-script/tests/index.test.ts
@@ -46,7 +46,6 @@ async function startWranglerPagesDevProxy(extraArgs: string[] = []): Promise<{
 				port: parsedMessage.port,
 			});
 		});
-		debugger;
 	});
 }
 

--- a/fixtures/pages-dev-proxy-with-script/tests/index.test.ts
+++ b/fixtures/pages-dev-proxy-with-script/tests/index.test.ts
@@ -1,0 +1,78 @@
+import path from "node:path";
+import { Response, fetch } from "undici";
+import { describe, it, expect } from "vitest";
+import { ChildProcess, fork } from "node:child_process";
+import { setTimeout } from "node:timers/promises";
+
+describe("Pages dev with proxy and a script file", () => {
+	it("should handle requests using a script from the default _worker.js path", async () => {
+		const { port, childProcess } = await startWranglerPagesDevProxy();
+		const combinedResponse = await waitUntilReady(`http://127.0.0.1:${port}/`);
+		const respText = await combinedResponse.text();
+		expect(respText).toMatchInlineSnapshot('"hello from _worker.js"');
+		await terminateChildProcess(childProcess);
+	});
+
+	it("should handle requests using a script from a custom script path", async () => {
+		const { port, childProcess } = await startWranglerPagesDevProxy([
+			"--script-path=custom/script/path/index.js",
+		]);
+		const combinedResponse = await waitUntilReady(`http://127.0.0.1:${port}/`);
+		const respText = await combinedResponse.text();
+		expect(respText).toMatchInlineSnapshot(
+			'"hello from custom/script/path/index.js"'
+		);
+		await terminateChildProcess(childProcess);
+	});
+});
+
+async function startWranglerPagesDevProxy(extraArgs: string[] = []): Promise<{
+	childProcess: ChildProcess;
+	port: string;
+}> {
+	return new Promise(async (resolve) => {
+		const childProcess = fork(
+			path.join("..", "..", "packages", "wrangler", "bin", "wrangler.js"),
+			["pages", "dev", "--port=0", "--proxy=9999", ...extraArgs],
+			{
+				cwd: path.resolve(__dirname, ".."),
+				env: { BROWSER: "none", ...process.env },
+				stdio: ["ignore", "ignore", "ignore", "ipc"],
+			}
+		).on("message", (message) => {
+			const parsedMessage = JSON.parse(message.toString());
+			resolve({
+				childProcess,
+				port: parsedMessage.port,
+			});
+		});
+		debugger;
+	});
+}
+
+function terminateChildProcess(childProcess: ChildProcess): Promise<unknown> {
+	return new Promise((resolve, reject) => {
+		childProcess.once("exit", (code) => {
+			if (!code) {
+				resolve(code);
+			} else {
+				reject(code);
+			}
+		});
+		childProcess.kill("SIGTERM");
+	});
+}
+
+async function waitUntilReady(url: string): Promise<Response> {
+	let response: Response | undefined = undefined;
+
+	while (response === undefined) {
+		await setTimeout(500);
+
+		try {
+			response = await fetch(url);
+		} catch (e) {}
+	}
+
+	return response as Response;
+}

--- a/fixtures/pages-dev-proxy-with-script/tests/tsconfig.json
+++ b/fixtures/pages-dev-proxy-with-script/tests/tsconfig.json
@@ -1,0 +1,7 @@
+{
+	"extends": "@cloudflare/workers-tsconfig/tsconfig.json",
+	"compilerOptions": {
+		"types": ["node"]
+	},
+	"include": ["**/*.ts", "../../../node-types.d.ts"]
+}

--- a/fixtures/pages-dev-proxy-with-script/tsconfig.json
+++ b/fixtures/pages-dev-proxy-with-script/tsconfig.json
@@ -1,0 +1,12 @@
+{
+	"include": ["index.d.ts"],
+	"compilerOptions": {
+		"target": "ES2020",
+		"module": "CommonJS",
+		"lib": ["ES2020"],
+		"types": ["@cloudflare/workers-types"],
+		"moduleResolution": "node",
+		"noEmit": true,
+		"skipLibCheck": true
+	}
+}

--- a/fixtures/pages-dev-proxy-with-script/vitest.config.ts
+++ b/fixtures/pages-dev-proxy-with-script/vitest.config.ts
@@ -1,0 +1,10 @@
+import { defineConfig } from "vitest/config";
+
+export default defineConfig({
+	test: {
+		testTimeout: 10_000,
+		hookTimeout: 10_000,
+		teardownTimeout: 10_000,
+		useAtomics: true,
+	},
+});

--- a/packages/wrangler/src/pages/dev.ts
+++ b/packages/wrangler/src/pages/dev.ts
@@ -291,7 +291,7 @@ export const Handler = async ({
 	const workerScriptPath =
 		directory !== undefined
 			? join(directory, singleWorkerScriptPath)
-			: singleWorkerScriptPath;
+			: resolve(singleWorkerScriptPath);
 	const usingWorkerDirectory =
 		existsSync(workerScriptPath) && lstatSync(workerScriptPath).isDirectory();
 	const usingWorkerScript = existsSync(workerScriptPath);

